packages/webdriverio/src/utils/Timer.ts

```diff
@@ -29,14 +33,14 @@ class Timer {
             this._fn = () => runFnInFiberContext(fn)()
         }
 
-        const retPromise = new Promise((resolve, reject) => {
+        const retPromise = new Promise<Timer>((resolve, reject) => {
             this._resolve = resolve
             this._reject = reject
         })
 
         this.start()
 
-        return retPromise
+        return { ...this, retPromise }
```

Comment:
Think it's ok to spread the context here? Otherwise types will complain that `retPromise` is missing several properties from the class (which makes sense)

Reply:
This was wrong, nevermind